wstl/pswstl2milp.py

```diff
@@ -0,0 +1,359 @@
+'''
+ Copyright (C) 2018-2020 Cristian Ioan Vasile <cvasile@lehigh.edu>
+ Hybrid and Networked Systems (HyNeSs) Group, BU Robotics Lab, Boston University
+ Explainable Robotics Lab, Lehigh University
+ See license.txt file for license information.
+ @author: Gustavo A. Cardona, Cristian-Ioan Vasile
+'''
+import gurobipy as grb
+
+from stl import Operation, RelOperation, STLFormula
+
+class pswstl2milp(object):
+    '''Translate an wSTL formula to an MILP.'''
+    def __init__(self, formula, ranges=None, vtypes=None, model=None, 
+                robust=False):
+        self.formula = formula
+        self.robust = robust
+        self.M = 1000
+        self.ranges = ranges
+        if ranges is None:
+            self.ranges = {v: (-10, 10) for v in self.formula.variables()}
+
+        assert set(self.formula.variables()) <= set(self.ranges)
+
+        self.vtypes = vtypes
+        if vtypes is None:
+            self.vtypes = {v: grb.GRB.CONTINUOUS for v in self.ranges}
+
+        self.model = model
+        if model is None:
+            self.model = grb.Model('pswSTL formula: {}'.format(formula))
+            
+        self.variables = dict()
+        self.__milp_call = {
+            Operation.PRED : self.predicate,
+            Operation.AND : self.conjunction,
+            Operation.EAND : self.conjunction,
+            Operation.OR : self.disjunction,
+            Operation.EOR : self.disjunction,
+            Operation.EVENT : self.eventually,
+            Operation.EEVENT : self.eventually,
+            Operation.ALWAYS : self.globally,
+            Operation.EALWAYS : self.globally,
```

Comment:
The exclusive operators (EAND, EOR, EEVENT, EALWAYS) do not seem to be defined in `Operation` class from `stl.py`.